src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,180 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            addFirst(value);
+        } else {
+            addAfter(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (isEmpty() && index == 0) {
+            addFirst(value);
+        } else if (index == 0) {
+            addBefore(value);
+        } else if (index == size) {
```

Comment:
the code can be simlified a bit if put
```
if (index == size) {
            add(value);
            return;
        }
```
to the begin of the class. 
the rest of the code can be done simple this way:
```   
     if (index == 0) {
        ... 
     } else {
        ...
     }